src/test/java/org/opensearch/sdk/TestExtensionSettings.java

```diff
@@ -0,0 +1,33 @@
+package org.opensearch.sdk;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class TestExtensionSettings extends OpenSearchTestCase {
+
+    private ExtensionSettings extensionSettings;
+
+    @BeforeEach
```

Comment:
It looks like `extensionSettings` is modified during the tests, this seems like it could lead to unpredictable behavior, I would recommend to reinitialize the value before every test run with the `@BeforeEach` annotation

Reply:
Well in the older behavior it was modified but only each individual field so they wouldn't step on each other.

And now it's an immutable object so we're creating an identical object three times.